lib/active_merchant/billing/gateways/d_local.rb

```diff
@@ -0,0 +1,226 @@
+        true
+      end
+
+      def scrub(transcript)
+        transcript.
+          gsub(%r((X-Trans-Key: )\w+), '\1[FILTERED]').
+          gsub(%r((\"number\\\":\\\")\d+), '\1[FILTERED]').
+          gsub(%r((\"cvv\\\":\\\")\d+), '\1[FILTERED]')
+      end
+
+      private
+
+      def add_auth_purchase_params(post, money, card, action, options)
+        add_invoice(post, money, options)
+        post[:payment_method_id] = 'CARD'
+        post[:payment_method_flow] = 'DIRECT'
+        add_country(post, card, options)
+        add_payer(post, card, options)
+        add_card(post, card, action, options)
+        post[:order_id] = options[:order_id] || generate_unique_id
+        post[:description] = options[:description] if options[:description]
+      end
+
+      def add_invoice(post, money, options)
+        post[:amount] = amount(money)
+        post[:currency] = (options[:currency] || currency(money))
+      end
+
+      def add_country(post, card, options)
+        return unless address = options[:billing_address] || options[:address]
+        post[:country] = lookup_country_code(address[:country])
+      end
+
+      def lookup_country_code(country)
+        Country.find(country).code(:alpha2)
+      end
+
+      def add_payer(post, card, options)
+        address = options[:billing_address] || options[:address]
+        post[:payer] = {}
+        post[:payer][:name] = card.name
+        post[:payer][:email] = options[:email] if options[:email]
+        post[:payer][:birth_date] = options[:birth_date] if options[:birth_date]
+        post[:payer][:phone] = address[:phone] if address[:phone]
+        post[:payer][:document] = options[:document] if options[:document]
+        post[:payer][:document2] = options[:document2] if options[:document2]
+        post[:payer][:user_reference] = options[:user_reference] if options[:user_reference]
+        post[:payer][:address] = add_address(post, card, options)
+      end
+
+      def add_address(post, card, options)
+        return unless address = options[:billing_address] || options[:address]
+        address_object = {}
+        address_object[:state] = address[:state] if address[:state]
+        address_object[:city] = address[:city] if address[:city]
+        address_object[:zip_code] = address[:zip_code] if address[:zip_code]
+        address_object[:street] = address[:street] if address[:street]
+        address_object[:number] = address[:number] if address[:number]
+        address_object
+      end
+
+      def add_card(post, card, action, options={})
+        post[:card] = {}
+        post[:card][:holder_name] = card.name
+        post[:card][:expiration_month] = card.month
+        post[:card][:expiration_year] = card.year
+        post[:card][:number] = card.number
+        post[:card][:cvv] = card.verification_value
+        post[:card][:descriptor] = options[:dynamic_descriptor] if options[:dynamic_descriptor]
+        post[:card][:capture] = (action == 'purchase')
```

Comment:
Should `post[:card][:capture]` always be passed?

Reply:
I think it defaults to `false`, but there isn't much of a difference.